students/A_Zach/lesson05/mailroom3.py

```diff
@@ -0,0 +1,118 @@
+import sys
+from operator import itemgetter
+#Functions
+
+#Show list of names
+def show_list():
+    #print names of donors
+    names = list(donor_dict)
+    name_string = "\n".join(names)
+    print(f"{name_string}")
+
+
+#make the program quit
+def quit_prog():
+    print("Thank you. Goodbye!")
+    sys.exit()
+
+#generate an email1
+def write_email(name):
+    email = f"\n\nDear {name.title()},\n\n\tWe appreciate your generous donations totalling ${sum(donor_dict[name.title()]):.2f}.\n\nThank you,\nAndrew\n\n" 
+    with open(f"Thank_You_to_{name.title()}.txt", 'w+') as f:
+        f.write(email)
+
+    
+def send_thank_you():
+    name = input("Who would you like to thank?>")
+    if name.lower() == 'quit':
+        return
+    
+    #show the list of names using show_list function
+    while name.title() == 'List':
+        show_list()
+        name = input("Who whould you like to thank?>")
+        if name.lower() == 'quit':
+            return
+    
+    donation = input("Please enter an amount to donate >")
+    while True:
+        #call quit_prog function to end program
+        if donation.lower() == 'quit':
+            return
+        try:
+            donor_dict.setdefault(name.title(),[]).append(float(donation))
+        except ValueError:
+            donation = input(f"\nInvalid response.\n\nEnter a number.\n\nPlease enter an amount to donate >")
+        else:
+            break
+    #call the email function
+    write_email(name)
+    return
+
+def thank_you_all():
+    [write_email(name) for name in donor_dict]
```

Comment:
this is an incorrect use of a list comprehension. right now you are populating a list of None values since write_email has no return value.

List comprehensions should only be used for cases when you are generating values to be stored in a list.

Here it should just be a regular for loop.